wagon.py

```diff
@@ -0,0 +1,1015 @@
+    method assumes tar.gz when installing on Windows as well.
+
+    Package name and version are extracted from the setup.py file
+    of the `source` or from the PACKAGE_NAME==PACKAGE_VERSION if the source
+    is a PyPI package.
+
+    Excluded packages will be removed from the archive even if they are
+    required for installation and their wheel names will be appended
+    to the metadata for later analysis/validation.
+
+    Supported `python_versions` must be in the format e.g [33, 27, 2, 3]..
+
+    `force` will remove any excess dirs or archives before creation.
+
+    `with_requirements` can be either a link/local path to a
+    requirements.txt file or just `.`, in which case requirement files
+    will be automatically extracted from either the GitHub archive URL
+    or the local path provided provided in `source`.
+    """
+    logger.info('Creating archive for {0}...'.format(source))
+    original_source = source
+    source, remove_source_after_process = get_source(source)
+    package_name, package_version = get_source_name_and_version(source)
+
+    excluded_packages = excluded_packages or []
+    if excluded_packages:
+        logger.warn('Note that excluding packages may make the archive '
+                    'non-installable.')
+    if package_name in excluded_packages:
+        raise WagonError(
+            'You cannot exclude the package you are trying to wheel.')
+
+    tempdir = tempfile.mkdtemp()
+    workdir = os.path.join(tempdir, package_name)
+    wheels_path = os.path.join(workdir, DEFAULT_WHEELS_PATH)
+
+    if with_requirements:
+        with_requirements = _get_default_requirement_files(source)
+
+    try:
+        wheels, excluded_wheels = wheel(
+            source, with_requirements, wheels_path, excluded_packages,
+            wheel_args)
+    finally:
+        if remove_source_after_process:
+            shutil.rmtree(source, ignore_errors=True)
+
+    platform = _get_platform_for_set_of_wheels(wheels_path)
+    logger.debug('Platform is: {0}'.format(platform))
+    python_versions = _set_python_versions(python_versions)
+
+    if not os.path.isdir(archive_destination_dir):
+        os.makedirs(archive_destination_dir)
+    archive_name = _set_archive_name(
+        package_name, package_version, python_versions, platform)
+    archive_path = os.path.join(archive_destination_dir, archive_name)
+
+    _handle_output_file(archive_path, force)
+    _generate_metadata_file(
+        workdir,
+        archive_name,
+        platform,
+        python_versions,
+        package_name,
+        package_version,
+        original_source,
+        wheels,
+        excluded_wheels)
+
+    _create_wagon_archive(format, workdir, archive_path)
+    if not keep_wheels:
+        logger.debug('Cleaning up...')
+        shutil.rmtree(workdir, ignore_errors=True)
+
+    if validate_archive:
+        validate(archive_path, verbose)
+    logger.info('Process complete!')
```

Comment:
I suggest to log out a more descriptive text like "Wagon archive created successfully at {0}".format(archive_path)